x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +235,136 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
+        } else if (EnrichPolicy.RANGE_TYPE.equals(policy.getType())) {
+            resolveRangeTypeEnrichMapping(policy, getIndexResponse, resultListener);
         } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
+            createEnrichMappingBuilder((builder) -> builder.field("type", "geo_shape"), resultListener);
         } else {
-            throw new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType());
+            resultListener.onFailure(new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType()));
         }
+    }
 
+    private void resolveRangeTypeEnrichMapping(
+        EnrichPolicy policy,
+        GetIndexResponse response,
+        ActionListener<XContentBuilder> resultListener
+    ) {
+        String matchField = policy.getMatchField();
+        List<String> types = collapseFieldMappingsValue(response, toFieldMappingProperty(matchField, "type"));
+        if (types.isEmpty()) {
+            resultListener.onFailure(
+                new ElasticsearchException(
+                    "No mapping type found for match field '{}' - indices({})",
+                    policy.getMatchField(),
+                    Strings.collectionToCommaDelimitedString(policy.getIndices())
+                )
+            );
+        } else if (types.size() > 1) {
+            resultListener.onFailure(
+                new ElasticsearchException(
+                    "Multiple distinct mapping types for match field '{}' - indices({})  types({})",
+                    policy.getMatchField(),
+                    Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                    Strings.collectionToCommaDelimitedString(types)
+                )
+            );
+        } else {
+            String matchType = resolveMatchType(types.get(0));
+            if (matchType == null) {
+                resultListener.onFailure(
+                    new ElasticsearchException(
+                        "Field '{}' has type [{}] which doesn't appear to be a range type",
+                        policy.getMatchField(),
+                        types.get(0)
+                    )
+                );
+            } else {
+                // date_range types mappings allow for the format to be specified, should be preserved in the created index
+                if (matchType.equals("date_range")) {
+                    List<String> formatEntries = collapseFieldMappingsValue(response, toFieldMappingProperty(matchField, "format"));
+                    if (formatEntries.isEmpty()) {
+                        createEnrichMappingBuilder(
+                            (builder) -> builder.field("type", matchType).field("doc_values", false),
+                            resultListener
+                        );
+                    } else if (formatEntries.size() > 1) {
```

Comment:
I think the `format` option in field mappers can only just have a single value. It should be possible that a date range mapping has more than 1 date format strings. (note that multiple date formats can be specified inside a single value using the `||` notation)

So I don't think this needs to be checked?

Reply:
You can have multiple source indicies, therefor I assumed there can be distinct mappings that might conflict, is there something else stopping this?

> indices
> (Required, String or array of strings) One or more source indices used to create the enrich index.
Honestly I haven't seen it get used in the field, but we seem to allow it.

Reply:
I see, I haven't thought about that. I think it is for now best to fail with an error if that were to happen. 

Although this code could also combine the different formats (from different date_range mappings from different indices) into a single date format using `||` notation. But let's keep it as is know, this could always be changed later.

Reply:
Check, I also had [this](https://github.com/elastic/elasticsearch/pull/76110#issuecomment-907542635) comment a little higher on this issue.